packages/ckeditor5-table/src/tablelayout/tablelayoutediting.ts

```diff
@@ -0,0 +1,212 @@
+	ViewElement
+} from 'ckeditor5/src/engine.js';
+
+import { createEmptyTableCell } from '../utils/common.js';
+import { scanTable } from '../converters/upcasttable.js';
+
+import InsertTableLayoutCommand from './../commands/inserttablelayoutcommand.js';
+
+/**
+ * The table layout editing plugin.
+ */
+export default class TableLayoutEditing extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	public static get pluginName() {
+		return 'TableLayoutEditing' as const;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public static override get isOfficialPlugin(): true {
+		return true;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public init(): void {
+		this._defineSchema();
+		this._defineConverters();
+
+		this.editor.commands.add( 'insertTableLayout', new InsertTableLayoutCommand( this.editor ) );
+	}
+
+	/**
+	 * Defines the schema for the table layout feature.
+	 */
+	private _defineSchema() {
+		const { schema } = this.editor.model;
+
+		schema.extend( 'table', {
+			allowAttributes: 'tableType'
+		} );
+
+		// Disallow adding caption to layout table.
+		schema.addChildCheck( context => {
+			if ( context.endsWith( 'table' ) && context.last.getAttribute( 'tableType' ) == 'layout' ) {
+				return false;
+			}
+		}, 'caption' );
+	}
+
+	/**
+	 * Defines the converters for the table layout feature.
+	 */
+	private _defineConverters() {
+		const { editor } = this;
+		const { conversion } = editor;
+
+		conversion.for( 'upcast' ).add( upcastTableOverride() );
+		conversion.for( 'upcast' ).add( upcastTableToSetTableType() );
+
+		editor.conversion.for( 'dataDowncast' ).add( dispatcher => {
+			return dispatcher.on( 'attribute:tableType:table', ( evt, data, conversionApi ) => {
+				const { item, attributeNewValue } = data;
+				const { mapper, writer } = conversionApi;
+
+				if ( !conversionApi.consumable.test( item, evt.name ) ) {
+					return;
+				}
+
+				const table = mapper.toViewElement( item );
+
+				writer.addClass( `${ attributeNewValue }-table`, table );
+				writer.setAttribute( 'role', 'presentation', table );
+
+				conversionApi.consumable.consume( item, evt.name );
+			} );
+		} );
+
+		conversion.for( 'editingDowncast' ).add( dispatcher => {
+			dispatcher.on( 'attribute:tableType', ( evt, data, conversionApi ) => {
+				const { item, attributeNewValue } = data;
+				const { mapper, writer } = conversionApi;
+				const modelElement = item;
+				const viewElement = mapper.toViewElement( modelElement );
+
+				writer.addClass( `${ attributeNewValue }-table`, viewElement );
+			} );
+		} );
+	}
+}
+
+/**
+ * View table element to model table element conversion helper.
+ *
+ * This conversion helper overrides the default table converter to meet table layout conditions.
+ *
+ * @returns Conversion helper.
+ */
+function upcastTableOverride() {
+	return ( dispatcher: UpcastDispatcher ): void => {
+		dispatcher.on<UpcastElementEvent>( 'element:table', (
+			evt: EventInfo,
+			data: UpcastConversionData<ViewElement>,
+			conversionApi: UpcastConversionApi
```

Comment:
```suggestion
			data: UpcastConversionData<ViewElement>,
			conversionApi: UpcastConversionApi
```
Are those types required or are infered from the `<UpcastElementEvent>`?